platform/linux-dpdk/odp_packet_dpdk.c

```
@@ -625,12 +637,21 @@ static int setup_pkt_dpdk(odp_pktio_t pktio ODP_UNUSED,
 	struct rte_eth_dev_info dev_info;
 	pkt_dpdk_t * const pkt_dpdk = pkt_priv(pktio_entry);
 	int i;
+	uint16_t port_id;
 
-	if (!_dpdk_netdev_is_valid(netdev)) {
-		ODP_DBG("Interface name should only contain numbers!: %s\n", netdev);
-		return -1;
+	if (_dpdk_netdev_is_pci_addr(netdev)) {
+		if (rte_eth_dev_get_port_by_name(netdev, &port_id)) {
+			ODP_DBG("Invalid PCI address for DPDK!: %s\n", netdev);
+			return -1;
+		}
+		pkt_dpdk->port_id = port_id;
+	} else {
+		if (!_dpdk_netdev_is_valid(netdev)) {
+			ODP_DBG("Interface name should only contain numbers!: %s\n", netdev);
+			return -1;
+		}
+		pkt_dpdk->port_id = atoi(netdev);
```

This is a good fix and I think we can improve it further.  

According to `rte_eth_dev_get_port_by_name()` documentation there are also other valid device names than a PCIe address (SoC device name, vdev name).

In odp-dpdk there are only three pktio types (`_odp_pktio_if_ops`): loopback, dpdk, and null. If you would change the order of the pktio types, so that dpdk would be the last one, this device name check code could be made simpler and more flexible. You could always call `rte_eth_dev_get_port_by_name()` without having to do the `_dpdk_netdev_is_pci_addr()` check. E.g. if `rte_eth_dev_get_port_by_name()` fails -> call `_dpdk_netdev_is_valid()`, and if both fail return error.

---

@MatiasElo If there is no other significance for the device order in _odp_pktio_if_ops I can move _odp_pktio_if_ops to the end. This will remove the need of parsing the name.
Also does the same apply for the device order for the same in io_ops.c, is it ok to move _odp_dpdk_pktio_ops to the end to fix the same issue in dpdk_open() for linux_generic ?

---

> If there is no other significance for the device order in _odp_pktio_if_ops I can move _odp_pktio_if_ops to the end. This will remove the need of parsing the name.

It only affects to the order in which pktios are tested. Loopback and null pktios will fail silently if the name doesn't match, so dpdk pktio can safely be moved to be the last one.

> Also does the same apply for the device order for the same in io_ops.c, is it ok to move _odp_dpdk_pktio_ops to the end to fix the same issue in dpdk_open() for linux_generic ?

In linux-generic we have multiple pktio types and some other type could accept the device name before dpdk pktio, so I wouldn't change `io_ops.c:_odp_pktio_if_ops` order. But I guess you can still use the same fix there. If the netdev is not a valid dpdk device name, dpdk pktio open can simply fail and linux-generic will just continue and try the next pktio type in _odp_pktio_if_ops.

---

Changes updated in v2 for odp-dpdk.
Will create patch as discussed for odp